### PR TITLE
ADD: Dev Menu (only for debug mode)

### PR DESCRIPTION
**Pull request by marcosrdz**
This menu will allow me to trigger UI changes based on providers without leaving the screens in question 

Simulator Screenshot - iPhone 15 Pro Max - 2024-08-25 at 10 44 57
Simulator Screenshot - iPhone 15 Pro Max - 2024-08-25 at 10 45 14

## Summary by CodeRabbit

- **New Features**
	- Introduced a developer menu for enhanced wallet management during development.
	- Added functionality to dynamically set and retrieve screen size values.

- **Improvements**
	- Refactored the way screen size detection is handled for better clarity and extensibility.
	- Improved the structure of hooks and context for managing screen size, allowing for more flexible use.

- **Bug Fixes**
	- Ensured the developer menu functionality is only accessible in development mode to prevent duplication.

- **Documentation**
	- Updated internal documentation related to new components and hooks.

### CODE REVIEW

File: components/DevMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+import { HDSegwitBech32Wallet } from '../class';
+import Clipboard from '@react-native-clipboard/clipboard';
+import { useIsLargeScreen } from '../hooks/useIsLargeScreen';
+import { TWallet } from '../class/wallets/types';
+
+const getRandomLabelFromSecret = (secret: string): string => {
+  const words = secret.split(' ');
+  const randomWord = words[Math.floor(Math.random() * words.length)];
+  return `[Developer] ${randomWord}`;
+};
```
Comment (Overtorment) [inline components/DevMenu.tsx:13]: 1st & last words are more memorable tbh, and consistent across multiple runs

File: components/DevMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      });
+
+      DevSettings.addMenuItem('Reset Screen Interface', () => {
+        setLargeScreenValue(undefined);
+        Alert.alert('Screen Interface reset to default.');
+      });
+    }
+  }, [wallets, addWallet, setLargeScreenValue]);
+
+  return null;
```
Comment (Overtorment) [inline components/DevMenu.tsx:183]: a lot of unnecessary actions. alsosome actions are duplicated in wallet/details screen (like tx purge). remove

Comment (Overtorment) [reply components/DevMenu.tsx:183]: leave only the ones you personally really need

**Review by Overtorment — CHANGES_REQUESTED** [review]
pls for your prs write a bit of rationale & howtos.

for example, i dont know how to trigger a dev menu?

also i dont know why we need it? definately not for sake of just having a dev menu.
every developers backdoors exist to solve some specific developers pains.

File: components/DevMenu.tsx (hunk 2 above)
Comment (marcosrdz) [reply components/DevMenu.tsx:183]: These are needed to make sure every relevant screen auto updates. Useful for bug reports of "this screen requires me to leave and return for it to update". Especially on large screen devices. All of these are the essentials..

**Comment by marcosrdz** [comment]
> pls for your prs write a bit of rationale & howtos.
> 
> for example, i dont know how to trigger a dev menu?
> 
> also i dont know why we need it? definately not for sake of just having a dev menu. every developers backdoors exist to solve some specific developers pains.

https://reactnative.dev/docs/debugging#:~:text=React%20Native%20provides%20an%20in,%2B%20M%20(Windows%20and%20Linux)

Those are the keyboard shortcuts. I want these so I can obtain the info I need quickly without having to click around buttons and screens. Especially context updates.

File: components/DevMenu.tsx (hunk 1 above)
Comment (marcosrdz) [reply components/DevMenu.tsx]: adjusted

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by Overtorment — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**